Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -340,12 +341,31 @@ void ImageBitmap::createCompletionHandler(ScriptExecutionContext& scriptExecutio
         return;
     }
 
+    createCompletionHandler(scriptExecutionContext, cachedImage, imageElement->renderer(), WTFMove(options), rect, WTFMove(completionHandler));
+}
+
+void ImageBitmap::createCompletionHandler(ScriptExecutionContext& scriptExecutionContext, RefPtr<SVGImageElement>& imageElement, ImageBitmapOptions&& options, std::optional<IntRect> rect, ImageBitmapCompletionHandler&& completionHandler)
+{
+    // 2. If image is not completely available, then return a promise rejected with
+    // an "InvalidStateError" DOMException and abort these steps.
+
+    auto* cachedImage = imageElement->cachedImage();
+    if (!cachedImage) {
+        completionHandler(Exception { InvalidStateError, "Cannot create ImageBitmap that is not completely available"_s });
+        return;
+    }
```

Comment:
Maybe this part can be moved to `createCompletionHandler(ScriptExecutionContext, CachedImage*,...)`. So it will not be duplicated here and in `createCompletionHandler(ScriptExecutionContext&, RefPtr<HTMLImageElement>&...)`.

```
 createCompletionHandler(scriptExecutionContext, imageElement->cachedImage(), imageElement->renderer(), WTFMove(options), rect, WTFMove(completionHandler));
```

Reply:
Good idea, will do!